### PR TITLE
Fix overlord scheduled executor injection

### PR DESCRIPTION
**Pull request by drcrallen**
* Fixes https://github.com/druid-io/druid/issues/2308

### CODE REVIEW

**Comment by fjy** [comment]
@drcrallen can we add a test?

**Comment by himanshug** [comment]
:+1: UT would be nice to have but i would expect that the fix is manually verified by running overlord.

**Comment by pjain1** [comment]
:+1: overlord starts fine with this fix

File: indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java [hunk 1]
```diff
@@ -79,7 +77,7 @@ public RemoteTaskRunnerFactory(
     this.cleanupExec = factory.create(1, "RemoteTaskRunner-Scheduled-Cleanup--%d");
     this.config = config;
     this.resourceManagementSchedulerConfig = resourceManagementSchedulerConfig;
-    this.exec = exec;
+    this.exec = factory.create(1, "RemoteTaskRunner-ResourceManagement--%d");
```
Comment (gianm) [inline indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java]: [minor] Would prefer to change the name of this to `resourceManagementExec`

**Comment by gianm** [comment]
:+1:

File: indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java (hunk 1 above)
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java]: I'd like to fix master ASAP. Will put in another pr with proposed change.

**Comment by drcrallen** [comment]
@himanshug / @fjy Yes, I was looking into how to test it. Currently there aren't really any tests for starting the nodes other than integration tests. So a test might not be as readily available as I would like.

**Comment by fjy** [comment]
@drcrallen @gianm travis is borked: https://www.traviscistatus.com/

Do we want to merge this to fix master?

**Comment by drcrallen** [comment]
@fjy build is running now. Suggest giving it another 30 mins to see if it completes

**Comment by gianm** [comment]
I'm ok merging without travis, given they are having an outage. the tests pass for me locally.

**Comment by drcrallen** [comment]
@fjy green

File: indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java (hunk 1 above)
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java]: https://github.com/druid-io/druid/pull/2311
